Chess/movevalidatorv2.cpp

```diff
@@ -0,0 +1,32 @@
+#include "movevalidatorv2.h"
+#include "board.h"
+
+
+MoveValidatorV2::MoveValidatorV2() {}
+
+bool MoveValidatorV2::validateMove(int diffRow, int diffCol, int row, int column)
+{
+    Piece* movedPiece = Board::getInstance()->getPieceForPosition(row, column);
+    ChessRulesValidator* rulesValidator = getChessRuleValidator(movedPiece);
```

Comment:
Nu prea inteleg dc este nevoie de un vector<rules*> daca mereu este creat la inceput functiei si dupa distrus la finalul ei.
vector<rules*> nu va contine mereu 1 singur item/regula?
Am putea avea practic un rules * in loc de vector<rules*>

Reply:
Momentan da, avem o singura regula, dar ulterior vom adauga reguli, iar fiecare piesa va trebui sa respecte propriul set de reguli (de ex: pionul nu va avea voie sa mearga inapoi, o piesa nu va putea fi mutata daca va lasa regele in sah)
Urmeaza sa adaugam si restul regulilor